eZ/Publish/Core/Repository/Repository.php

```diff
@@ -259,6 +264,7 @@ public function __construct(
         PasswordHashServiceInterface $passwordHashGenerator,
         ThumbnailStrategy $thumbnailStrategy,
         ProxyDomainMapperFactoryInterface $proxyDomainMapperFactory,
+        LoginTypeResolver $loginTypeResolver,
```

Comment:
Q: Why not simply pass which credentials type are enabled using `$serviceSettings`? IMHO that would simplify implementation a lot :-)

Reply:
Because which login type is enabled and which is not, may be dynamic and may depend on various circumstances. Implementation doesn't have to be simple if it doesn't complicate usage. This way it's more flexible.